src/main/java/genesis/Genesis.java

```diff
@@ -0,0 +1,184 @@
+        if (tasks.size() < 1) {
+            throw new GenesisException("List is empty, nothing to show");
+        }
+
+        ConsolePrinter.listTasks(tasks);
+    }
+
+    private void handleMarkTask(String content) {
+        int index = Integer.parseInt(content) - 1;
+        Task task = tasks.get(index);
+        task.setIsDone(true);
+
+        ConsolePrinter.onTaskMarked(task.getListDescription());
+    }
+
+    private void handleUnmarkTask(String content) {
+        int index = Integer.parseInt(content) - 1;
+        Task task = tasks.get(index);
+        task.setIsDone(false);
+
+        ConsolePrinter.onTaskUnmarked(task.getListDescription());
+    }
+
+    private void handleTodo(String description) throws GenesisException {
+        Todo todo = new Todo(description);
+        tasks.add(todo);
+
+        ConsolePrinter.onTaskAdded(todo.getListDescription(), tasks.size());
+    }
+
+    private void handleDeadline(String content) throws GenesisException {
+        String[] parts = content.split(" /by ", 2);
+        if (parts.length < 2) {
+            throw new GenesisException("Description for deadline is invalid");
+        }
+
+        String description = parts[0];
+        String by = parts[1];
+
+        Deadline deadline = new Deadline(description, by);
+        tasks.add(deadline);
+
+        ConsolePrinter.onTaskAdded(deadline.getListDescription(), tasks.size());
+    }
+
+    private void handleEvent(String content) throws GenesisException {
+        String[] parts = content.split(" /from | /to ", 3);
+        if (parts.length < 3) {
+            throw new GenesisException("Description for event is invalid");
+        }
+
+        String description = parts[0];
+        String from = parts[1];
+        String to = parts[2];
+
+        Event event = new Event(description, from, to);
+        tasks.add(event);
+
+        ConsolePrinter.onTaskAdded(event.getListDescription(), tasks.size());
+    }
+
+    private void handleDelete(String content) {
+        int index = Integer.parseInt(content) - 1;
+        Task task = tasks.remove(index);
+
+        ConsolePrinter.onTaskDelete(task.getListDescription(), tasks.size());
+    }
+
+
+    private void validateIndex(String[] contentArr) throws GenesisException {
+
+        if (contentArr.length < 2) {
+            throw new GenesisException("Index cannot be empty");
+        }
+    }
+
+    private void validateDescription(String[] contentArr) throws GenesisException {
+        if (contentArr.length < 2 || contentArr[1].isBlank()) {
+            throw new GenesisException("The description cannot be empty.");
+        }
+    }
+
+    public void askGenesis(String userInput) {
+        try {
+            ConsolePrinter.breakLine();
+
+            String[] contentArr = userInput.split(" ", 2);
+            String command = contentArr[0];
+
+            switch (command) {
+            case "list": {
+                handleListTasks();
+                return;
```

Comment:
`break`?

Reply:
My idea was to exit the method completely after handleListTasks() is completed. 
That is why i chose to use return instead of break so that it will not run the method FileHandler.saveToFile(tasks)